Documentation/general/cli-ux-guidelines.md

```diff
@@ -0,0 +1,155 @@
+    items)? If yes, then it should be added as a noun to the “dotnet
+    add/list/remove/update”, e.g. “dotnet add foo”.
+
+-   Does the command work on the solution (SLN) file? If yes, then it should be
+    added as a verb to the “dotnet sln” command.
+
+-   Does the command work on a completely new artifact (e.g. a new metadata file
+    that is added in the future)? If yes, it should be created as a top level
+    noun with all of the underlying operations defined as sub-commands.
+
+-   If the command adds a new artifact to the project, it should become an item
+    template that is dropped with “dotnet new” command.
+
+This setup was adopted to provide a clear namespacing that the users can
+remember and that fits well with usage patterns observed in other CLI toolsets a
+well as Visual Studio.
+
+If none of the above applies, the proposal should clearly outline why none of
+the above applies and suggest an alternative naming that will be decided on
+during the process described above.
+
+If the command works on the project or solution, it must also accept an optional
+argument that specifies which project or solution to work on. The current
+convention is for that argument to follow the verb in the CRUD command. For
+example:
+
+>   dotnet add \<PROJECT\> reference \<REFERENCE\>
+
+All the existing CRUD commands have this argument defined and it will be passed
+to the sub-command as part of the overall options. The sub-command is expected
+to consider this optional argument.
+
+## Options 
+CLI follows the GNU convention for options’ format, which is based on the POSIX
+standard. The summary is:
+
+-   Command can have both short and long options.
+
+-   The short format starts with a single dash (“-“), and has **exactly** one
+    letter. The letter is usually the start letter of the longer form of the
+    option, e.g. “-o” for “--output” or any of its words. This provides a good
+    mnemonic.
+
+    -   **Example:** “dotnet build -o path/to/where/to/put/output”
+
+-   The long format starts with double dashes (“--“) and can have one or more
+    words. Multiple words are separated by a single dash (“-“).
+
+    -   **Example:** “dotnet test --test-case-filter”
+
+-   The double dashes (“--“) on their own mark the **argument separator**.
+    Anything after this option is passed verbatim to whatever the command starts
+    as a sub-process.
+
+    -   **Example:** “dotnet run -- --app-arg-1”
+
+-   Windows-style forward slash options (e.g. “/p”) are supported **only for
+    MSBuild parameters.**
+
+-   PowerShell-style single dashes with a work (e.g. “-Filter”) are **not
+    supported**.
+
+-   Help options are predefined as “-h \| --help” and those should be used.
+
+There are common options in the CLI that are reserved for certain concepts. The
+table below outlines those common options. If a command needs to accept the same
+options, it should use these and it must not override their semantics (e.g. use

```

Comment:
I wouldn't use the term "override" here, but maybe "replace" or "alter" or something that. It isn't really "override".